StepDefinitions/SkillFeatureStepDefinitions.cs

```diff
@@ -0,0 +1,107 @@
+using MarsOnboardingIC.Pages;
+using MarsOnboardingIC.Utilities;
+using NUnit.Framework;
+using OpenQA.Selenium;
+using Reqnroll;
+using System;
+using Assert = NUnit.Framework.Assert;
+
+namespace MarsOnboardingIC.StepDefinitions
+{
+    [Binding]
+    public class SkillFeatureStepDefinitions
+    {
+        public IWebDriver driver;
+        public SkillFeatureStepDefinitions()
+        {
+            // Get driver instance from Hooks
+            driver = Hooks.driver;
+        }
+        [Given("I logged in as a register user")]
+        public void GivenILoggedInAsARegisterUser()
+        {
+
+            LoginPage loginPageObj = new LoginPage();
+            loginPageObj.LoginAction(driver);
```

Comment:
Please do not pass the driver In action methods (methods which in pages doing things like create, delete etc ) instead get the pages to extend to common driver where you will create a common driver class which will initialise the web driver 

public class CommonDriver
    {
        public static IWebDriver driver;

        public void initialise()
        {
            // create driver
            driver = new ChromeDriver();

        }

    }
and then in the hooks  create a common driver instance which calls the initialise method 

5/ 
please make use of Hooks , you might need to create the Common driver object there!
Please see
https://codebun.com/how-to-implement-hooks-in-specflow-c/